adyenv6b2ccheckoutaddonv2/acceleratoraddon/web/src/com/adyen/v6/api/AdyenDeliveryAddressContoller.java

```diff
@@ -0,0 +1,89 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import javax.annotation.Resource;
+
+@Controller
+@RequestMapping(value = "/api/checkout")
+public class AdyenDeliveryAddressContoller {
+
+    @Resource(name = "checkoutFacade")
+    private CheckoutFacade checkoutFacade;
+
+    @Resource(name = "userFacade")
+    private UserFacade userFacade;
+
+    @PostMapping(value = "/delivery-address")
+    @RequireHardLogIn
+    public ResponseEntity<HttpStatus>  doSelectDeliveryAddress(@RequestBody final String selectedAddressCode)
+    {
+        if (StringUtils.isNotBlank(selectedAddressCode))
+        {
+            final AddressData selectedAddressData = getCheckoutFacade().getDeliveryAddressForCode(selectedAddressCode);
+            final boolean hasSelectedAddressData = selectedAddressData != null;
+            if (hasSelectedAddressData)
+            {
+                setDeliveryAddress(selectedAddressData);
+                return ResponseEntity.status(HttpStatus.ACCEPTED).build();
+            }
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+        }
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
+    }
+
+    @GetMapping(value = "/delivery-address")
+    @RequireHardLogIn
+    public ResponseEntity<AddressData> getSelectedDeliveryAddress()
+    {
+        final AddressData selectedAddressData = getCheckoutFacade().getCheckoutCart().getDeliveryAddress();
+        if (selectedAddressData != null)
+        {
+            return ResponseEntity.status(HttpStatus.OK).body(selectedAddressData);
+        }
+        return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+    }
+
+    protected void setDeliveryAddress(final AddressData selectedAddressData) {
+        final AddressData cartCheckoutDeliveryAddress = getCheckoutFacade().getCheckoutCart().getDeliveryAddress();
+        if (isAddressIdChanged(cartCheckoutDeliveryAddress, selectedAddressData)) {
+            getCheckoutFacade().setDeliveryAddress(selectedAddressData);
+            if (cartCheckoutDeliveryAddress != null && !cartCheckoutDeliveryAddress.isVisibleInAddressBook()) { // temporary address should be removed
+                getUserFacade().removeAddress(cartCheckoutDeliveryAddress);
+            }
+        }
+    }
+
+    protected boolean isAddressIdChanged(final AddressData cartCheckoutDeliveryAddress, final AddressData selectedAddressData)
+    {
+        return cartCheckoutDeliveryAddress == null || !selectedAddressData.getId().equals(cartCheckoutDeliveryAddress.getId());
+    }
+
+
+    public UserFacade getUserFacade() {
+        return userFacade;
```

Comment:
Czy jeśli używamy @Resource to potrzebujemy getterów i setterów?

Reply:
When you use @Resource on a field, the dependency is injected directly into that field. In this case, getters and setters are not strictly necessary for dependency injection itself, but is nothing wrong with using it in case we will want to maintain the flexibility for future inheritance